perl/imap/IMAP.xs

```diff
@@ -717,26 +717,23 @@ imclient_toURL(client,server,box)
         char *server
         char *box
 PREINIT:
-        char *out_buf;
-        int len;
+        struct buf buf = BUF_INITIALIZER;
         struct imapurl imapurl;
 PPCODE:
-        len = strlen(server)+strlen(box);
-        out_buf = safemalloc(4*len);
-
         memset(&imapurl, 0, sizeof(struct imapurl));
         imapurl.server = server;
         imapurl.mailbox = box;
-        imapurl_toURL(out_buf, &imapurl);
+        imapurl_toURL(&buf, &imapurl);
+        buf_cstring(&buf);
 
-        if(!out_buf[0]) {
-                safefree(out_buf);
+        if(!buf_len(&buf)) {
+                buf_free(&buf);
                 XSRETURN_UNDEF;
         }
 
-        XPUSHs(sv_2mortal(newSVpv(out_buf, 0)));
+        XPUSHs(sv_2mortal(newSVpv(buf_release(&buf), 0)));
 
         /* newSVpv copies this */
-        safefree(out_buf);
+        buf_free(&buf);
```

Comment:
Though actually, the comment here says that `newSVpv` _copies_ its argument, and the original `out_buf` was then freed, supporting that claim.

In which case, I think it's fine to use `newSVpv()` with the malloced memory from `struct buf`, because it doesn't take ownership -- but then the argument should be `buf_cstring(&buf)`, not `buf_release(&buf)`, otherwise the buffer will be ownerless and leaked.